File: VBNetCore.Validators/TextValidator.vb
Imports System.Collections.Generic

Namespace VBNetCore.Validators
    Public Class TextValidator
        Private ReadOnly _errorMessages As Dictionary(Of String, String) = New Dictionary(Of String, String)()
        Private Shared ReadOnly _instance As TextValidator = New TextValidator()
        Private Sub New()
            _errorMessages.Add("Required", "{0} Can not be empty.")
        End Sub
        Public Shared ReadOnly Property Instance As TextValidator
            Get
                Return _instance
            End Get
        End Property
        Public Function Required(ByVal Text As String) As Boolean
            If String.IsNullOrEmpty(Text) OrElse String.IsNullOrWhiteSpace(Text) OrElse Equals(Text, String.Empty) OrElse Equals(Text, " ") Then
                Return False
            Else
                Return True
            End If
        End Function
        Public Function ErrorMessage(ByVal ruleName As String, ByVal propertyName As String) As String
            Return If(_errorMessages.ContainsKey(ruleName), String.Format(_errorMessages(ruleName), propertyName), String.Empty)
        End Function
    End Class
End Namespace

File: VBNetCore.Validators/TextValidator.cs
using System.Collections.Generic;

namespace NetCore.Validators
{
    public class TextValidator
    {
        private readonly Dictionary<string, string> _errorMessages = new Dictionary<string, string>();
        private static readonly TextValidator _instance = new TextValidator();
        private TextValidator()
        {
            _errorMessages.Add("Required", "{0} Can not be empty.");
        }
        public static TextValidator Instance => _instance;
        public bool Required(string? Text)
        {
            if (string.IsNullOrEmpty(Text) || string.IsNullOrWhiteSpace(Text) || Text == string.Empty || Text == " ")
            {
                return false;
            }
            else return true;
        }
        public string ErrorMessage(string ruleName, string propertyName)
        {
            return _errorMessages.ContainsKey(ruleName) ? string.Format(_errorMessages[ruleName], propertyName) : string.Empty;
        }
    }
}
